client/my-sites/migrate/section-migrate.js

```diff
@@ -39,36 +65,160 @@ class SectionMigrate extends Component {
+			} );
+	};
+
+	renderLoading() {
+		return (
+			<CompactCard>
+				<span className="migrate__placeholder">Loading...</span>
+			</CompactCard>
+		);
+	}
+
+	renderMigrationComplete() {
+		const { targetSite } = this.props;
+		const viewSiteURL = get( targetSite, 'URL' );
+
+		return (
+			<CompactCard>
+				<div className="migrate__status">Your migration has completed successfully.</div>
+				<Button primary href={ viewSiteURL }>
+					View site
+				</Button>
+				<Button onClick={ this.resetMigration }>Start over</Button>
+			</CompactCard>
+		);
+	}
+
+	renderMigrationConfirmation() {
+		const { sourceSite, targetSite, translate } = this.props;
+
+		const sourceSiteDomain = get( sourceSite, 'domain' );
+		const targetSiteDomain = get( targetSite, 'domain' );
+
+		return (
+			<CompactCard>
+				<div className="migrate__confirmation">
+					{ translate(
+						'Do you want to migrate all content from %(sourceSite)s to %(targetSite)s? All existing content on %(sourceSite)s will be lost.',
+						{ args: { sourceSite: sourceSiteDomain, targetSite: targetSiteDomain } }
+					) }
+				</div>
+				<Button primary onClick={ this.startMigration }>
+					{ translate( 'Start Migration' ) }
+				</Button>
+			</CompactCard>
+		);
+	}
+
+	renderMigrationError() {
+		return (
+			<CompactCard>
+				<div className="migrate__status">There was an error with your migration.</div>
+				<Button primary onClick={ this.resetMigration }>
+					Try again
+				</Button>
+			</CompactCard>
+		);
+	}
+
+	renderMigrationStatus() {
+		const { migrationStatus } = this.state;
+
+		return <div>{ migrationStatus }</div>;
+	}
+
+	renderSourceSiteSelector() {
 		const { translate } = this.props;
+
+		return (
+			<CompactCard className="migrate__card">
+				<SiteSelector
+					className="migrate__source-site"
+					selected={ this.state.sourceSiteId }
+					onSiteSelect={ this.setSourceSiteId }
+					filter={ this.jetpackSiteFilter }
+				/>
+				<Button primary onClick={ this.selectSourceSite } disabled={ ! this.state.sourceSiteId }>
+					{ translate( 'Continue' ) }
+				</Button>
+			</CompactCard>
+		);
+	}
+
+	render() {
+		const { sourceSiteId, translate } = this.props;
 		const headerText = translate( 'Migrate' );
 		const subHeaderText = translate( 'Migrate your WordPress site to WordPress.com' );
 
+		let migrationElement;
+
+		switch ( this.state.migrationStatus ) {
+			case 'inactive':
+				migrationElement = sourceSiteId
+					? this.renderMigrationConfirmation()
+					: this.renderSourceSiteSelector();
+				break;
+
+			case 'restoring':
+				migrationElement = this.renderMigrationStatus();
+				break;
+
+			case 'done':
+				migrationElement = this.renderMigrationComplete();
+				break;
+
+			case 'error':
+				migrationElement = this.renderMigrationError();
+				break;
+
+			case 'unknown':
+			default:
+				migrationElement = this.renderLoading();
+		}
+
 		return (
 			<Main>
+				<Interval onTick={ this.updateFromAPI } period={ EVERY_FIVE_SECONDS } />
```

Comment:
Given the speed of migrations, this seems too often. Maybe every minute? Unless we ever see migrations take less than a minute.